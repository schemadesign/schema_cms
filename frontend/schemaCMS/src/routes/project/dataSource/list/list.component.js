import React, { PureComponent } from 'react';

import { Container } from './list.styles';

export class List extends PureComponent {
  static propTypes = {};

  render() {
    return <Container>List component</Container>;
  }
}
