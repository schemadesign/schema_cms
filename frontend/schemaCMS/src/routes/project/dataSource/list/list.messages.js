/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'dataSource.list.title',
    defaultMessage: 'Data Sources',
  },
  subTitle: {
    id: 'dataSource.list.subTitle',
    defaultMessage: 'Overview',
  },
  projectDetails: {
    id: 'dataSource.list.projectDetails',
    defaultMessage: 'Project details',
  },
  error: {
    id: 'dataSource.list.error',
    defaultMessage: 'Error',
  },
  processing: {
    id: 'dataSource.list.processing',
    defaultMessage: 'Loading...',
  },
  draft: {
    id: 'dataSource.list.draft',
    defaultMessage: 'Draft',
  },
  source: {
    id: 'dataSource.list.source',
    defaultMessage: 'Source',
  },
  items: {
    id: 'dataSource.list.items',
    defaultMessage: 'Items',
  },
  fields: {
    id: 'dataSource.list.fields',
    defaultMessage: 'Fields',
  },
  filters: {
    id: 'dataSource.list.filters',
    defaultMessage: 'Filters',
  },
  views: {
    id: 'dataSource.list.views',
    defaultMessage: 'Views',
  },
});
