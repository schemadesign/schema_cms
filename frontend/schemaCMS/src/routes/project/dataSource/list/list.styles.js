import styled, { css } from 'styled-components';
import { Theme } from 'schemaUI';
import { styleWhenTrue } from '../../../../shared/utils/rendering';

export const Container = styled.div`
  margin: 0 20px 60px 20px;
`;

export const DataSourceItem = styled.li`
  margin-top: 10px;

  &:first-child {
    margin-top: 0;
  }
`;

export const Description = styled.div`
  margin: 7px 0 14px;
  cursor: pointer;
`;

export const DataSourceList = styled.div`
  list-style: none;
  padding: 0;
  margin-top: 24px;
`;

export const titleStyles = {
  cursor: 'pointer',
  marginBottom: '20px',
};

export const MetaDataWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const MetaData = styled.div`
  display: flex;
  flex-direction: column;
`;

export const MetaDataName = styled.span`
  color: ${Theme.primary.label};
  font-weight: 600;
  font-size: 12px;
  margin-bottom: 10px;
`;

const lockStyles = styleWhenTrue(
  ({ isLock }) => isLock,
  css`
    color: ${Theme.primary.label};
  `
);

export const MetaDataValue = styled.span`
  font-weight: 600;
  font-size: 24px;
  line-height: 36px;

  ${lockStyles}
`;

export const Header = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const HeaderIcon = styled.div`
  margin-top: -8px;
`;

export const lockTextStyles = {
  color: Theme.primary.label,
};

export const addDataSourceStyles = {
  position: 'fixed',
  right: '10px',
  bottom: '10px',
  backgroundColor: Theme.primary.label,
  height: '60px',
};

export const iconSourceStyles = {
  width: 40,
  height: 40,
  fill: Theme.primary.iconDark,
};

export const lockIconStyles = {
  fill: Theme.primary.label,
};
