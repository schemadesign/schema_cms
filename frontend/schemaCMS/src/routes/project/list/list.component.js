import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button, Card, Header, Icons, Menu, Typography } from 'schemaUI';
import { isEmpty } from 'ramda';
import moment from 'moment';

import { renderWhenTrueOtherwise } from '../../../shared/utils/rendering';
import {
  Action,
  ActionsList,
  Container,
  Description,
  Empty,
  HeaderItem,
  HeaderList,
  ProjectItem,
  ProjectsList,
  headerStyles,
  urlStyles,
  addProjectStyles,
} from './list.styles';

const { H1, H2, P, Span } = Typography;

export class List extends PureComponent {
  static propTypes = {
    list: PropTypes.array.isRequired,
    fetchProjectsList: PropTypes.func.isRequired,
    isMenuOpen: PropTypes.bool.isRequired,
    toggleMenu: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.fetchProjectsList();
  }

  handleToggleMenu = () => {
    this.props.toggleMenu();
  };

  handleNewProject = () => {};

  renderHeader = (list = []) => (
    <HeaderList>
      {list.map((item, index) => (
        <HeaderItem key={index}>{item}</HeaderItem>
      ))}
    </HeaderList>
  );

  renderItem({ id, title = '', description = '', slug = '', created = '', status = '', owner = {} }) {
    const { firstName = '', lastName = '' } = owner;
    const user = isEmpty(firstName) ? lastName : `${firstName} ${lastName}`;
    const whenCreated = moment(created).fromNow();

    const header = this.renderHeader([whenCreated, status, user]);

    return (
      <ProjectItem key={id}>
        <Card headerComponent={header}>
          <H1>{title}</H1>
          <Description>
            <P>{description}</P>
          </Description>
          <Span customStyles={urlStyles}>{slug}</Span>
        </Card>
      </ProjectItem>
    );
  }

  renderList = (_, list) => <ProjectsList>{list.map(item => this.renderItem(item))}</ProjectsList>;

  renderNoData = () => (
    <Empty>
      <P>No Projects</P>
    </Empty>
  );

  renderMenu = () => {
    const { isMenuOpen } = this.props;

    return (
      <Menu open={isMenuOpen} onClose={this.handleToggleMenu}>
        <ActionsList>
          <Action>Edit Project settings</Action>
          <Action>Delete project</Action>
          <Action>User settings</Action>
          <Action>Log Out</Action>
        </ActionsList>
      </Menu>
    );
  };

  render() {
    const { list = [] } = this.props;
    const content = renderWhenTrueOtherwise(this.renderList, this.renderNoData)(Boolean(list.length), list);

    return (
      <Container>
        <Header customStyles={headerStyles} onButtonClick={this.handleToggleMenu}>
          <H2>Projects</H2>
          <H1>Overview</H1>
        </Header>
        {content}
        <Button customStyles={addProjectStyles} onClick={this.handleNewProject}>
          <Icons.PlusIcon />
        </Button>
        {this.renderMenu()}
      </Container>
    );
  }
}
