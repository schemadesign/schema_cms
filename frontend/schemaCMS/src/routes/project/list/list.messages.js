/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'project.list.pageTitle',
    defaultMessage: 'Projects Overview',
  },
  title: {
    id: 'project.list.title',
    defaultMessage: 'Projects',
  },
  overview: {
    id: 'project.list.overview',
    defaultMessage: 'Overview',
  },
  noProjects: {
    id: 'project.list.noProjects',
    defaultMessage: 'No Projects',
  },
});
