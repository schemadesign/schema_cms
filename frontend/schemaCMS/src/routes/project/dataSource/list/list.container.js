import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { injectIntl } from 'react-intl';

import { compose } from 'ramda';
import { List } from './list.component';
import { DataSourceRoutines } from '../../../../modules/dataSource';

const mapStateToProps = createStructuredSelector({});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      createDataSource: promisifyRoutine(DataSourceRoutines.create),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  injectIntl,
  withRouter
)(List);
