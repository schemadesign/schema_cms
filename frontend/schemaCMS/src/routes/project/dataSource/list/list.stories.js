import React from 'react';
import { storiesOf } from '@storybook/react';

import { List } from './list.component';

const defaultProps = {
  createDataSource: Function.prototype,
  match: {
    params: {
      projectId: '1',
    },
  },
};

storiesOf('Project/DataSource/List', module).add('Default', () => <List {...defaultProps} />);
