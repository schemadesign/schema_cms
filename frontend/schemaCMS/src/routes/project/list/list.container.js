import { connect } from 'react-redux';
import { promisifyRoutine, bindPromiseCreators } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { injectIntl } from 'react-intl';
import { compose } from 'ramda';

import { List } from './list.component';
import { ProjectRoutines } from '../../../modules/project';
import { selectProjectsList } from '../../../modules/project/project.selectors';

const mapStateToProps = createStructuredSelector({
  list: selectProjectsList,
});

export const mapDispatchToProps = dispatch =>
  bindPromiseCreators(
    {
      fetchProjectsList: promisifyRoutine(ProjectRoutines.fetchList),
      unmountProject: promisifyRoutine(ProjectRoutines.unmountOne),
    },
    dispatch
  );

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  injectIntl,
  withRouter
)(List);
