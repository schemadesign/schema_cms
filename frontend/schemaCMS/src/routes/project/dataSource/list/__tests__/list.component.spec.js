import React from 'react';
import { shallow } from 'enzyme';

import { List } from '../list.component';
import {
  STATUS_DRAFT,
  STATUS_ERROR,
  STATUS_PROCESSING,
  STATUS_READY_FOR_PROCESSING,
} from '../../../../../modules/dataSource/dataSource.constants';

describe('List: Component', () => {
  const defaultProps = {
    createDataSource: Function.prototype,
    fetchDataSources: Function.prototype,
    cancelFetchListLoop: Function.prototype,
    dataSources: [],
    history: {
      push: Function.prototype,
    },
    match: {
      params: {
        projectId: '1',
      },
    },
    intl: {
      formatMessage: ({ defaultMessage }) => defaultMessage,
    },
  };

  const dataSource = {
    created: '2019-09-09T11:23:40+0000',
    createdBy: { firstName: 'firstName', lastName: 'lastName' },
    id: 17,
    metaData: {
      fields: 11,
      items: 246,
    },
    name: 'name',
    status: 'done',
  };

  const component = props => <List {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render empty', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const props = {
      dataSources: [dataSource],
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render card with error state', () => {
    dataSource.status = STATUS_ERROR;
    dataSource.errorLog = ['error 1', 'error2'];
    const props = {
      dataSources: [dataSource],
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render card with draft state', () => {
    dataSource.status = STATUS_DRAFT;
    const props = {
      dataSources: [dataSource],
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render card with processing state', () => {
    dataSource.status = STATUS_PROCESSING;
    const props = {
      dataSources: [dataSource],
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render card with ready for processing state', () => {
    dataSource.status = STATUS_READY_FOR_PROCESSING;
    const props = {
      dataSources: [dataSource],
    };

    const wrapper = render(props);
    global.expect(wrapper).toMatchSnapshot();
  });
});
