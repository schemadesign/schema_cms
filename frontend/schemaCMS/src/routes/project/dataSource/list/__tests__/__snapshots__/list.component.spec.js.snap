// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List: Component should render card with draft state 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Sources"
  />
  <TopHeader
    headerSubtitle="Overview"
    headerTitle="Data Sources"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Project details",
          "to": "/project/view/1",
        },
      ]
    }
  />
  <styled.div>
    <styled.li
      key="0"
    >
      <Card
        headerComponent={
          <FormattedMessage
            defaultMessage="Draft"
            id="dataSource.list.draft"
            values={Object {}}
          />
        }
      >
        <H1
          customStyles={
            Object {
              "color": "#a6a7aa",
              "cursor": "pointer",
              "marginBottom": "20px",
            }
          }
          onClick={[Function]}
        >
          name
        </H1>
        <styled.div>
          <styled.div
            key="0"
          >
            <styled.span>
              Source
            </styled.span>
            <styled.span
              isLock={true}
            >
              <CsvIcon
                customStyles={
                  Object {
                    "fill": "#a6a7aa",
                    "height": 40,
                    "width": 40,
                  }
                }
              />
            </styled.span>
          </styled.div>
          <styled.div
            key="1"
          >
            <styled.span>
              Items
            </styled.span>
            <styled.span
              isLock={true}
            >
              246
            </styled.span>
          </styled.div>
          <styled.div
            key="2"
          >
            <styled.span>
              Fields
            </styled.span>
            <styled.span
              isLock={true}
            >
              11
            </styled.span>
          </styled.div>
          <styled.div
            key="3"
          >
            <styled.span>
              Filters
            </styled.span>
            <styled.span
              isLock={true}
            >
              —
            </styled.span>
          </styled.div>
          <styled.div
            key="4"
          >
            <styled.span>
              Views
            </styled.span>
            <styled.span
              isLock={true}
            >
              —
            </styled.span>
          </styled.div>
        </styled.div>
      </Card>
    </styled.li>
  </styled.div>
  <Button
    customStyles={
      Object {
        "backgroundColor": "#a6a7aa",
        "bottom": "10px",
        "height": "60px",
        "position": "fixed",
        "right": "10px",
      }
    }
    onClick={[Function]}
  >
    <PlusIcon />
  </Button>
</styled.div>
`;

exports[`List: Component should render card with error state 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Sources"
  />
  <TopHeader
    headerSubtitle="Overview"
    headerTitle="Data Sources"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Project details",
          "to": "/project/view/1",
        },
      ]
    }
  />
  <styled.div>
    <styled.li
      key="0"
    >
      <Card
        headerComponent={
          <FormattedMessage
            defaultMessage="Error"
            id="dataSource.list.error"
            values={Object {}}
          />
        }
      >
        <H1
          customStyles={
            Object {
              "color": "#a6a7aa",
              "cursor": "pointer",
              "marginBottom": "20px",
            }
          }
          onClick={[Function]}
        >
          name
        </H1>
        <styled.div>
          <styled.div
            key="0"
          >
            error 1
          </styled.div>
          <styled.div
            key="1"
          >
            error2
          </styled.div>
        </styled.div>
      </Card>
    </styled.li>
  </styled.div>
  <Button
    customStyles={
      Object {
        "backgroundColor": "#a6a7aa",
        "bottom": "10px",
        "height": "60px",
        "position": "fixed",
        "right": "10px",
      }
    }
    onClick={[Function]}
  >
    <PlusIcon />
  </Button>
</styled.div>
`;

exports[`List: Component should render card with processing state 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Sources"
  />
  <TopHeader
    headerSubtitle="Overview"
    headerTitle="Data Sources"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Project details",
          "to": "/project/view/1",
        },
      ]
    }
  />
  <styled.div>
    <styled.li
      key="0"
    >
      <Card
        headerComponent={
          <FormattedMessage
            defaultMessage="Loading..."
            id="dataSource.list.processing"
            values={Object {}}
          />
        }
      >
        <H1
          customStyles={
            Object {
              "color": "#a6a7aa",
              "cursor": "pointer",
              "marginBottom": "20px",
            }
          }
          onClick={[Function]}
        >
          name
        </H1>
        <styled.div>
          <styled.div
            key="0"
          >
            <styled.span>
              Source
            </styled.span>
            <styled.span
              isLock={true}
            >
              <CsvIcon
                customStyles={
                  Object {
                    "fill": "#a6a7aa",
                    "height": 40,
                    "width": 40,
                  }
                }
              />
            </styled.span>
          </styled.div>
          <styled.div
            key="1"
          >
            <styled.span>
              Items
            </styled.span>
            <styled.span
              isLock={true}
            >
              246
            </styled.span>
          </styled.div>
          <styled.div
            key="2"
          >
            <styled.span>
              Fields
            </styled.span>
            <styled.span
              isLock={true}
            >
              11
            </styled.span>
          </styled.div>
          <styled.div
            key="3"
          >
            <styled.span>
              Filters
            </styled.span>
            <styled.span
              isLock={true}
            >
              —
            </styled.span>
          </styled.div>
          <styled.div
            key="4"
          >
            <styled.span>
              Views
            </styled.span>
            <styled.span
              isLock={true}
            >
              —
            </styled.span>
          </styled.div>
        </styled.div>
      </Card>
    </styled.li>
  </styled.div>
  <Button
    customStyles={
      Object {
        "backgroundColor": "#a6a7aa",
        "bottom": "10px",
        "height": "60px",
        "position": "fixed",
        "right": "10px",
      }
    }
    onClick={[Function]}
  >
    <PlusIcon />
  </Button>
</styled.div>
`;

exports[`List: Component should render card with ready for processing state 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Sources"
  />
  <TopHeader
    headerSubtitle="Overview"
    headerTitle="Data Sources"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Project details",
          "to": "/project/view/1",
        },
      ]
    }
  />
  <styled.div>
    <styled.li
      key="0"
    >
      <Card
        headerComponent={
          <FormattedMessage
            defaultMessage="Loading..."
            id="dataSource.list.processing"
            values={Object {}}
          />
        }
      >
        <H1
          customStyles={
            Object {
              "color": "#a6a7aa",
              "cursor": "pointer",
              "marginBottom": "20px",
            }
          }
          onClick={[Function]}
        >
          name
        </H1>
        <styled.div>
          <styled.div
            key="0"
          >
            <styled.span>
              Source
            </styled.span>
            <styled.span
              isLock={true}
            >
              <CsvIcon
                customStyles={
                  Object {
                    "fill": "#a6a7aa",
                    "height": 40,
                    "width": 40,
                  }
                }
              />
            </styled.span>
          </styled.div>
          <styled.div
            key="1"
          >
            <styled.span>
              Items
            </styled.span>
            <styled.span
              isLock={true}
            >
              246
            </styled.span>
          </styled.div>
          <styled.div
            key="2"
          >
            <styled.span>
              Fields
            </styled.span>
            <styled.span
              isLock={true}
            >
              11
            </styled.span>
          </styled.div>
          <styled.div
            key="3"
          >
            <styled.span>
              Filters
            </styled.span>
            <styled.span
              isLock={true}
            >
              —
            </styled.span>
          </styled.div>
          <styled.div
            key="4"
          >
            <styled.span>
              Views
            </styled.span>
            <styled.span
              isLock={true}
            >
              —
            </styled.span>
          </styled.div>
        </styled.div>
      </Card>
    </styled.li>
  </styled.div>
  <Button
    customStyles={
      Object {
        "backgroundColor": "#a6a7aa",
        "bottom": "10px",
        "height": "60px",
        "position": "fixed",
        "right": "10px",
      }
    }
    onClick={[Function]}
  >
    <PlusIcon />
  </Button>
</styled.div>
`;

exports[`List: Component should render correctly 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Sources"
  />
  <TopHeader
    headerSubtitle="Overview"
    headerTitle="Data Sources"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Project details",
          "to": "/project/view/1",
        },
      ]
    }
  />
  <styled.div>
    <styled.li
      key="0"
    >
      <Card
        headerComponent={
          <ForwardRef(styled.div)>
            <ForwardRef(styled.ul)>
              <ForwardRef(styled.li)>
                3 months ago
              </ForwardRef(styled.li)>
              <ForwardRef(styled.li)>
                firstName lastName
              </ForwardRef(styled.li)>
            </ForwardRef(styled.ul)>
            <ForwardRef(styled.div)>
              <IntersectIcon />
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>
        }
      >
        <H1
          customStyles={
            Object {
              "cursor": "pointer",
              "marginBottom": "20px",
            }
          }
          onClick={[Function]}
        >
          name
        </H1>
        <styled.div>
          <styled.div
            key="0"
          >
            <styled.span>
              Source
            </styled.span>
            <styled.span
              isLock={false}
            >
              <CsvIcon
                customStyles={
                  Object {
                    "fill": "#1d1d20",
                    "height": 40,
                    "width": 40,
                  }
                }
              />
            </styled.span>
          </styled.div>
          <styled.div
            key="1"
          >
            <styled.span>
              Items
            </styled.span>
            <styled.span
              isLock={false}
            >
              246
            </styled.span>
          </styled.div>
          <styled.div
            key="2"
          >
            <styled.span>
              Fields
            </styled.span>
            <styled.span
              isLock={false}
            >
              11
            </styled.span>
          </styled.div>
          <styled.div
            key="3"
          >
            <styled.span>
              Filters
            </styled.span>
            <styled.span
              isLock={false}
            >
              —
            </styled.span>
          </styled.div>
          <styled.div
            key="4"
          >
            <styled.span>
              Views
            </styled.span>
            <styled.span
              isLock={false}
            >
              —
            </styled.span>
          </styled.div>
        </styled.div>
      </Card>
    </styled.li>
  </styled.div>
  <Button
    customStyles={
      Object {
        "backgroundColor": "#a6a7aa",
        "bottom": "10px",
        "height": "60px",
        "position": "fixed",
        "right": "10px",
      }
    }
    onClick={[Function]}
  >
    <PlusIcon />
  </Button>
</styled.div>
`;

exports[`List: Component should render empty 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Data Sources"
  />
  <TopHeader
    headerSubtitle="Overview"
    headerTitle="Data Sources"
    secondaryMenuItems={
      Array [
        Object {
          "label": "Project details",
          "to": "/project/view/1",
        },
      ]
    }
  />
  <styled.div />
  <Button
    customStyles={
      Object {
        "backgroundColor": "#a6a7aa",
        "bottom": "10px",
        "height": "60px",
        "position": "fixed",
        "right": "10px",
      }
    }
    onClick={[Function]}
  >
    <PlusIcon />
  </Button>
</styled.div>
`;
